script.module.clouddrive.common/clouddrive/common/service/player.py

```diff
@@ -60,27 +62,91 @@ def stop(self):
 
 class KodiPlayer(KodiUtils.kodi_player_class()):
     def __init__(self, *args):
-        pass
+        self.iskrypton = KodiUtils.get_home_property('iskrypton') == 'true'
 
     def set_source_url_matcher(self, source_url_matcher):
         self.source_url_matcher = source_url_matcher
+    
+    def set_addonid(self, addonid):
+        self.addonid = addonid
         
     def onPlayBackStarted(self):
```

Comment:
For leia please use onAVStarted

Reply:
The problem is I need a callback function that works in both versions. I think I can change it sometime after leia is finally released.